### PR TITLE
Add support files

### PR DESCRIPTION
**Pull request by bombaywalla**
This is my first cut at adding community support files.  In particular, I've added
* `CONTRIBUTING.md` about how to make contributions to this repo.
* `CHANGELOG.md` to document changes. 
* `RELEASE.md` to document how to make releases.

I've tried to retrieve the changes made in the past and incorporate them into the `CHANGELOG.md`, but they may be incomplete.

My understanding of Github releases is rudimentary, so please verify that what I have said is correct.

Any comments or suggestions for improvement are welcome.

I have not yet added a PR template, but wanted to get these published first, so that we can get some experiences with the release process.

### CODE REVIEW

**Comment by bombaywalla** [comment]
I am somewhat concerned that the release instructions are too heavyweight - in that they try to do a release via PRs.  Perhaps there is a more expedient way?  Feedback appreciated.

Also, not sure how the releases to clojars are done currently.  I believe @yogthos has done the last few.

File: CONTRIBUTING.md
Comment (holyjak) [inline CONTRIBUTING.md:18]: I'd drop this one. You can do changes directly on GitHub, f.ex. to a Readme, without ever having to clone to a machine.

File: RELEASE.md
Comment (holyjak) [inline RELEASE.md:44]: I do not see why we would need to require the use of a fork

File: RELEASE.md
Comment (holyjak) [inline RELEASE.md:51]: I have no strong opinions but I would not mind keeping the process more lightweight and not requiring others to review. A release in my view is typically quite a small change. Perhaps you could ask for a review and if nobody responds within few hours / a day, you could proceed anyway?

File: RELEASE.md
Comment (holyjak) [inline RELEASE.md:57]: I don't think we need to bother with GH releases since our primary distribution is via Clojars. When you create a tag, it will be visible on the GH releases page automatically I believe so that perhaps satisfy those that are, for some reasons, interested in that.

**Review by holyjak — APPROVED** [review]
Great job! It looks +- good to me though I wouldn't mind making it more lightweight.

File: RELEASE.md
Comment (bombaywalla) [reply RELEASE.md:57]: We can hold off for now, but I do think that it is nice for people who browse Github to see what releases there have been.

File: RELEASE.md
Comment (bombaywalla) [reply RELEASE.md:51]: Fair enough.

I do think there is need for some sort of coordination when there are multiple maintainers.
But I'm probably over-thinking the process and it might be best leave things off here for now.

How about we create Slack thread or a Slack channel for `cryogen-dev`?  And we can coordinate there.

File: RELEASE.md
Comment (bombaywalla) [reply RELEASE.md:44]: I'm not sure I like the idea of making changes to files directly on Github.  I'd rather have PRs for better grouping of changes.  But perhaps these PRs can be merged by the author.  (As per your point in the comment below).

File: CONTRIBUTING.md
Comment (bombaywalla) [reply CONTRIBUTING.md:18]: I feel that all changes to the primary repo should be via PRs, and not via direct changes to files on Github.
This keeps all changes grouped properly (if changes are made to multiple files).

I'd prefer that the PRs come from the the individual's repo, not the main repo.
That way, the author is responsible for maintaining or removing the branches.
Otherwise the main repo will be cluttered with branches.

There might be exceptions in case of emergencies, but for the most part, I'd stick with the above.

Moreover, it seems that both of these conventions are what has been followed for this project in practice - once there were more than just one contributor.

Comment (holyjak) [reply CONTRIBUTING.md:18]: 1. Sorry, I have not explained myself properly. Even if you edit a file on GH, underneath GH will make a branch - and a fork if necessary - and submit a PR. So the process is the same, only the code never touches your machine.
2. "repo will be cluttered with branches" - that is a good argument indeed. Fortunately there is an automated solution - we can configure GH to automatically delete PR branches after a merge. 
3. To be fair, I should present my argument against mandatory forking: it adds extra steps - I have to remember to pull from upstream, not master, and reset to upstream - and from experience I know I mess/miss those up sometimes. Do you have any other arguments for mandatory forks?

File: RELEASE.md
Comment (holyjak) [reply RELEASE.md:44]: Explained in the previous comment thread.

File: RELEASE.md
Comment (holyjak) [reply RELEASE.md]: Why not just sticking with `cryogen`? But absolutely feel free to create it if you feel that is best.

File: RELEASE.md
Comment (holyjak) [reply RELEASE.md]: As mentioned, I believe tags will do that for us.

File: RELEASE.md
Comment (bombaywalla) [reply RELEASE.md]: OK.  Let's keep things as is for now.  Thanks.

File: CONTRIBUTING.md
Comment (bombaywalla) [reply CONTRIBUTING.md:18]: > Sorry, I have not explained myself properly. Even if you edit a file on GH, underneath GH will make a branch - and a fork if necessary - and submit a PR. So the process is the same, only the code never touches your machine.

I was not aware of this.  It would indeed be good if the code does not touch your machine.  Is this true only for people who have write access to the repo?

From [this documentation](https://docs.github.com/en/free-pro-team@latest/github/managing-files-in-a-repository/editing-files-in-your-repository) I see that Github will make a commit on `master` or create a specific branch on the repo itself.

>  we can configure GH to automatically delete PR branches after a merge.

That works for me.

Perhaps we can recommend (but not enforce) making a fork.  And perhaps we can mention that if making a direct change that a branch and PR is made.  Would that work?

Mostly, I am trying to continue the conventions that are already in place.  Just trying to document them and make them explicit.

File: RELEASE.md
Comment (bombaywalla) [reply RELEASE.md:44]: OK.

File: RELEASE.md
Comment (bombaywalla) [reply RELEASE.md]: Agreed.  I have removed the parts about making a Github release.

**Comment by bombaywalla** [comment]
I have pushed changes to this PR.  Please review and comment.  Thanks!

**Comment by bombaywalla** [comment]
If you think the current version is okay, would you please merge this PR to master?  Thanks.

File: CONTRIBUTING.md
Comment (holyjak) [reply CONTRIBUTING.md:18]: > I was not aware of this. It would indeed be good if the code does not touch your machine. Is this true only for people who have write access to the repo?
No, you can go to any OSS repo on GH and "edit" a file, which will create a fork and send PR.

> Perhaps we can recommend (but not enforce) making a fork. And perhaps we can mention that if making a direct change that a branch and PR is made. 
Yes!

**Review by holyjak — APPROVED** [review]

**Pull request merged**
